crates/cheatnet/src/state.rs

```diff
@@ -155,8 +163,40 @@ pub struct CheatnetState {
     pub block_info: BlockInfo,
     // execution resources used by all contract calls
     pub used_resources: UsedResources,
-    // trace info of the last call
-    pub trace_info: Vec<CallEntryPoint>,
+    /// Trace of calls made in a test.
+    /// The root `CallTrace` symbolizes calling the test code.
+    pub call_trace: Rc<RefCell<CallTrace>>,
+    /// Pointer to the call we are currently in.
+    pub current_call: Rc<RefCell<CallTrace>>,
+}
+
+impl Default for CheatnetState {
+    fn default() -> Self {
+        let call_trace = Rc::new(RefCell::new(CallTrace {
+            entry_point: build_test_entry_point(),
```

Comment:
Should we allow this? I think the entry point should be registered from outside, as a method preferably but i might be wrong here

Reply:
Yea I also don't like the idea of creating a purpose specific strucutre under `Default` trait. Imo it does not match the semantics of the trait.

Reply:
Please add this to description so it doesn't get lost